File: UI-Angular/src/models/registered-user.ts


export class RegisteredUser{
    id!: number;
    firstName: string;
    lastName: string;
    userName: string;
    password: string;
    email: string;
    role: RegisteredUserRole;
    phone: string;
    address: string;
    orderIds: Array<Number>;
    shoppingCartItemIds: Array<Number>;
    reviewIds: Array<Number>;
    
    constructor(firstName: string, lastName: string, userName: string, password: string, 
        email: string, role: RegisteredUserRole, phone: string, address: string, 
        orderIds: Array<Number>, shoppingCartItemIds: Array<Number>, reviewIds: Array<Number>){
            this.firstName = firstName,
            this.lastName = lastName,
            this.userName = userName,
            this.password = password,
            this.email = email,
            this.role = role,
            this.phone = phone,
            this.address = address,
            this.orderIds = orderIds,
            this.shoppingCartItemIds = shoppingCartItemIds,
            this.reviewIds = reviewIds;
        }



}

export enum RegisteredUserRole{
    RegisteredUser,
    WaterPointRepresentative,
    Admin
}

File: DataAccessLayer/Entities/RegisteredUser.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataAccessLayer.Entities
{
    public class RegisteredUser : BaseEntity
    {
        [Required, StringLength(60)]
        public string FirstName { get; set; }

        [Required, StringLength(60)]
        public string LastName { get; set; }

        [Required, StringLength(20)]
        public string UserName { get; set; }

        [Required, StringLength(20)]
        public string Password { get; set; }

        [Required]
        public RegisteredUserRole Role { get; set; }

        [Required, EmailAddress]
        public string Email { get; set; }

        [Required, StringLength(30)]
        public string Phone { get; set; }

        [Required, StringLength(80)]
        public string Address { get; set; }

        public List<Order> Orders { get; set; }
        public List<ShoppingCart> ShoppingCartItems { get; set;}
        public List<Review> Reviews { get; set; }
    }

    public enum RegisteredUserRole
    {
        RegisteredUser,
        WaterPointRepresentative,
        Admin
    }
}
